### PR TITLE
add regridding

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Add regridding for `mo_globa`l data

Changes reuslts from

Screenshot 2024-12-01 at 11 37 42

to

Screenshot 2024-12-01 at 11 42 42

## How Has This Been Tested?

- CI tests
- ran it locally, 

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: india_forecast_app/models/pvnet/utils.py [hunk 1]
```diff
@@ -134,12 +144,24 @@ def process_and_cache_nwp(source_nwp_path: str, dest_nwp_path: str):
...
+    elif nwp_config.source == "gfs":
         data_var = ds[list(ds.data_vars.keys())[0]]
         # # Use .to_dataset() to split the data variable based on 'variable' dim
         ds = data_var.to_dataset(dim="variable")
         ds = ds.rename({"t2m": "t"})
+
+    if nwp_config.source == "mo_global":
+
+        # only select the variables we need
+        ds = ds.sel(variable=["temperature_sl", "wind_u_component_10m", "wind_v_component_10m"])
```
Comment (Sukh-P) [inline india_forecast_app/models/pvnet/utils.py:156]: Hardcoding the variables here seems a bit risky, the weather variable selection is handled later on by datapipes/data-sampler so not sure if it is worth having this hardcoded here as well as there would be different variables for PV or wind

File: india_forecast_app/models/pvnet/utils.py [hunk 2]
```diff
@@ -115,10 +128,7 @@ def process_and_cache_nwp(source_nwp_path: str, dest_nwp_path: str):
         if ds[v].dtype == object:
             ds[v].encoding.clear()
 
-    is_gfs = "gfs" in source_nwp_path.lower()
-    is_ecmwf = "ecmwf" in source_nwp_path.lower()
-
-    if is_ecmwf:
+    if nwp_config.source == "ecmwf":
```
Comment (Sukh-P) [inline india_forecast_app/models/pvnet/utils.py:131]: Might be a separate PR but it would be good to do this for ECMWF as well as there is also a difference in spatial resolution between training and live for it, although it is worth noting that I think we the current HF config files we use it will error since we have hacked them to work and would need to change them back to the original number of pixels

File: india_forecast_app/data/nwp.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def regrid_nwp_data(nwp_ds: xr.Dataset, target_coords_path: str) -> xr.Dataset:
+    """This function loads the  NWP data, then regrids and saves it back out if the data is not
+    on the same grid as expected. The data is resaved in-place.
+
+    method can be 'conservative' or 'bilinear'
```
Comment (Sukh-P) [inline india_forecast_app/data/nwp.py:12]: What does this bit of the docstring refer to?

Comment (peterdudfield) [reply india_forecast_app/data/nwp.py:12]: oh, sorry, that can be removed now

File: india_forecast_app/models/pvnet/utils.py (hunk 2 above)
Comment (peterdudfield) [reply india_forecast_app/models/pvnet/utils.py:131]: yep, i was thinking a seperate PR for ECMWF stuff

File: india_forecast_app/models/pvnet/utils.py (hunk 1 above)
Comment (peterdudfield) [reply india_forecast_app/models/pvnet/utils.py:156]: yea good point. I can take them from the config though.

**Review by Sukh-P — APPROVED** [review]
